### PR TITLE
Metric Advisor: Adding support for updating api key

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/MetricsAdvisorKeyCredential.java [hunk 1]
```diff
@@ -38,4 +38,15 @@ public String getSubscriptionKey() {
...
         return this.apiKey;
     }
+
+    /**
+     * Rotates the api key associated to this credential.
+     *
+     * @param apiKey The new api key to associated with this credential.
+     * @return The updated {@code MetricsAdvisorKeyCredential} object.
+     */
+    public MetricsAdvisorKeyCredential updateApiKey(String apiKey) {
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/MetricsAdvisorKeyCredential.java:48]: I think we should call this `update` to be consistent with [AzureKeyCredential](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/credential/AzureKeyCredential.java#L50).

**Review by samvaity — APPROVED** [review]

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/MetricsAdvisorKeyCredential.java (hunk 1 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/MetricsAdvisorKeyCredential.java:48]: I was thinking about it a bit, the azure-key credential object compose only the key, but for MA, the credential type composes both subscription-key and api-key, so when we name it `update`, its not clear what property it is updating, additionally the getter is named as `getApiKey`. Let's bring it in our cross team sync.

Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/MetricsAdvisorKeyCredential.java:48]: C# is using `updateApiKey` [ref](https://github.com/Azure/azure-sdk-for-net/blob/ac24a13f2f99559c478675a75d2cf5f8174e5e3b/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorKeyCredential.cs#L65)

Comment (samvaity) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/MetricsAdvisorKeyCredential.java:66]: I think the suggestion from @schaabs is to have two separate methods `updateApiKey` and `updateSubscriptionKey`.

Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/MetricsAdvisorKeyCredential.java:66]: yep, that's exactly I did in this PR, I believe we're good

**Pull request merged**